File: JS Exams/01.  Basics Exam Collection/Basics Online Exam - 20 and 21 April 2019/04. Easter Eggs Battle/04. Easter Eggs Battle.js
function easterEggsBattle(...arg) {

    let eggsOnePlayer = arg.shift()
    let eggsTwoPlayer = arg.shift()

    while (true) {
        current = arg.shift()
        if (current == 'End of battle')
            return `Player one has ${eggsOnePlayer} eggs left.\nPlayer two has ${eggsTwoPlayer} eggs left.`

        current === 'one' ? eggsTwoPlayer-- : eggsOnePlayer--

        if (eggsTwoPlayer === 0) {
            return `Player two is out of eggs. Player one has ${eggsOnePlayer} eggs left.`
        } else if (eggsOnePlayer == 0) {
            return `Player one is out of eggs. Player two has ${eggsTwoPlayer} eggs left.`
        }
    }
}

//console.log(easterEggsBattle(5, 4, 'one', 'two', 'one', 'two', 'two', 'End of battle'))

File: Python Exams/01. Basic Exam Collection/Basics Online Exam - 20 and 21 April 2019/04. Easter Eggs Battle/04. Easter Eggs Battle.py
eggs_one_player = int(input())
eggs_two_player = int(input())

while True:
    current = input()
    if current == 'End of battle':
        print(f'Player one has {eggs_one_player} eggs left.\nPlayer two has {eggs_two_player} eggs left.')
        break

    if current == 'one':
        eggs_two_player -= 1
    elif current == 'two':
        eggs_one_player -= 1

    if eggs_two_player == 0:
        print(f'Player two is out of eggs. Player one has {eggs_one_player} eggs left.')
        break
    elif eggs_one_player == 0:
        print(f'Player one is out of eggs. Player two has {eggs_two_player} eggs left.')
        break
